src/application/worker/audio-features.js

```diff
@@ -1,13 +1,85 @@
+import Meyda from "meyda";
+import BeatDetektor from "../../../lib/BeatDetektor";
 import lerp from "../utils/lerp";
+import store from "./store";
 
 const MAX_SMOOTHING = 1;
 const SMOOTHING_STEP = 0.001;
 
 let features = {};
 const smoothedFeatures = {};
 
+let frame;
+let frameReader;
+let stream;
+
+let beatDetektor;
+let beatDetektorKick;
+
+const setupBeatDetektor = () => {
+  beatDetektor = new BeatDetektor(120, 180);
+  beatDetektorKick = new BeatDetektor.modules.vis.BassKick();
+};
+
+const updateBeatDetektor = (delta, features) => {
+  if (!features) {
+    return;
+  }
+  if (!beatDetektor) {
+    setupBeatDetektor();
+  }
+  beatDetektor.process(delta / 1000.0, features.complexSpectrum.real);
+  beatDetektorKick.process(beatDetektor);
+  const kick = beatDetektorKick.isKick();
+  const bpm = beatDetektor.win_bpm_int_lo;
+
+  store.commit("beats/SET_KICK", { kick });
+  if (store.state.beats.bpm !== bpm) {
+    store.dispatch("beats/setBpm", { bpm, source: "beatdetektor" });
+  }
+};
+
+function processFrame({ done, value: newFrame }) {
+  if (done) {
+    frameReader.releaseLock();
+    stream.cancel();
+    return;
+  }
+
+  frame = newFrame;
+}
+
 function getFeatures() {
-  return features;
+  return new Promise(resolve => {
+    const {
+      meyda: { features: featuresToGet }
+    } = store.state;
+
+    if (!frame) {
+      return;
+    }
+
+    let size = 0;
+    try {
+      size = frame.allocationSize({ planeIndex: 0 });
+    } catch (_) {
+      return;
+    }
+
+    Meyda.bufferSize = size;
+
+    const buffer = new ArrayBuffer(size);
+
```

Comment:
@TimPietrusky So to keep `features.byteFrequencyData = Array.from(getByteFrequencyData() || []);` we just do `features.byteFrequencyData = buffer;` I think?

Reply:
@TimPietrusky should be fixed in the latest commit.